modules/file-input/test/spec.coffee

```diff
@@ -16,116 +18,265 @@ describe "File Input", ->
     lastModified: new Date()
     type: 'image/png'
 
+  fakeEvent = {
+    # fake some dom event stuff
+    stopPropagation: ->
+  }
 
-  beforeEach ->
+  before ->
     hx.consoleWarning = chai.spy()
-    fixture.clear()
+    hx._.dropdown.attachToSelector = '#fixture'
+    clock = sinon.useFakeTimers()
 
-  afterEach ->
-    hx.consoleWarning = origConsoleWarning
+  beforeEach ->
+    hx.consoleWarning.reset()
+    fixture?.remove()
+    fixture = hx.select('body').append('div').attr('id', 'fixture')
 
   after ->
     fixture.remove()
+    hx.consoleWarning = origConsoleWarning
+    hx._.dropdown.attachToSelector = origDropdownAttachSelector
+    clock.restore()
+
+  fakeNodeEvent = (node, eventName) ->
+    if node?
+      (e) -> hx.select.getHexagonElementDataObject(node).eventEmitter?.emit((if eventName? and isNaN(eventName) then eventName else 'click'), e)
+
+  testFileInput = (options, callback) ->
+    sel = fixture.append('div')
+    fileInput = new hx.FileInput(sel.node(), options)
+    hx.consoleWarning.should.not.have.been.called()
+    callback(sel, fileInput)
+
+
+  it 'should have user facing text defined', ->
+    hx.userFacingText('fileInput','chooseFile').should.equal('Choose File')
+    hx.userFacingText('fileInput','chooseFiles').should.equal('Choose Files')
+    hx.userFacingText('fileInput','filesSelected').should.equal('Files Selected: $numFiles')
+    hx.userFacingText('fileInput','noFile').should.equal('No File Chosen')
+
 
-  emitEvent = (node, event, data) ->
-    hx.select.getHexagonElementDataObject(node).eventEmitter
-      .emit event, data
+  # Not sure how to test this?
+  # it 'should load a preview image', ->
```

Comment:
It uses a `FileReader` and I'm not sure how to create something that a file reader can read? I tried using a `Blob` but node doesn't have those...